components/server/src/github/github-repository-provider.ts

```diff
@@ -215,19 +215,20 @@ export class GithubRepositoryProvider implements RepositoryProvider {
               }`,
         );
 
-        let repos: RepositoryInfo[] = [];
-
+        const repos: RepositoryInfo[] = [];
         for (const type of ["contributedTo", "original", "forked"]) {
             const nodes = result.data.viewer[type]?.nodes;
             if (nodes) {
-                repos = nodes.map((n: any): RepositoryInfo => {
```

Comment:
This was actually a bug where we always rewrote the existing array if there were nodes for a type. Unlucky for us, even if there weren't, empty JS arrays are truthy, so the repos would get overiden anyway.